packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -205,16 +234,39 @@ void _writeIOSPluginRegistrant(String directory, List<Plugin> plugins) {
     'plugins': iosPlugins,
   };
 
-  final String pluginRegistryHeader =
-      new mustache.Template(_iosPluginRegistryHeaderTemplate).renderString(context);
-  final String pluginRegistryImplementation =
-      new mustache.Template(_iosPluginRegistryImplementationTemplate).renderString(context);
-  final Directory registryDirectory = fs.directory(fs.path.join(directory, 'ios', 'Runner'));
-  registryDirectory.createSync(recursive: true);
-  final File registryHeaderFile = registryDirectory.childFile('GeneratedPluginRegistrant.h');
-  registryHeaderFile.writeAsStringSync(pluginRegistryHeader);
-  final File registryImplementationFile = registryDirectory.childFile('GeneratedPluginRegistrant.m');
-  registryImplementationFile.writeAsStringSync(pluginRegistryImplementation);
+  if (manifest.isModule) {
+    // In a module create the GeneratedPluginRegistrant as a pod to be included
+    // from a hosting app.
+    final String registryDirectory = fs.path.join(directory, 'FlutterPluginRegistrant');
+    final String registryClassesDirectory = fs.path.join(registryDirectory, 'Classes');
+    _renderTemplateToFile(
+      _iosPluginRegistrantPodspecTemplate,
+      context,
+      fs.path.join(registryDirectory, 'FlutterPluginRegistrant.podspec'),
+    );
+    _renderTemplateToFile(
+      _iosPluginRegistryHeaderTemplate,
+      context,
+      fs.path.join(registryClassesDirectory, 'GeneratedPluginRegistrant.h'),
+    );
+    _renderTemplateToFile(
+      _iosPluginRegistryImplementationTemplate,
+      context,
+      fs.path.join(registryClassesDirectory, 'GeneratedPluginRegistrant.m'),
+    );
+  } else {
```

Comment:
Please add a comment about what the else clause does, similar to the one in the `if` branch.